addon/mixins/adapter-fetch.js

```diff
@@ -0,0 +1,142 @@
+import Ember from 'ember';
+import fetch from 'fetch';
+
+const { assign, RSVP } = Ember;
+
+/**
+ * Helper function that turns the data/body of a request into a query param string.
+ * @param {Object} queryParamsObject
+ * @returns {String}
+ */
+export function serialiazeQueryParams(queryParamsObject) {
+  return Object.keys(queryParamsObject).map((key) => {
+    return `${encodeURIComponent(key)}=${encodeURIComponent(queryParamsObject[key])}`;
+  }).join('&');
+}
+
+/**
+ * Helper function to create a plain object from the response's Headers.
+ * Consumed by the adapter's `handleResponse`.
+ * @param {Headers} headers
+ * @returns {Object}
+ */
+export function headersToObject(headers) {
+  let headersObject = {};
+  headers.forEach((value, key) => {
+    headersObject[key] = value;
+  });
+  return headersObject;
+}
+/**
+ * Helper function that translates the options passed to `jQuery.ajax` into a format that `fetch` expects.
+ * @param {Object} options
+ */
+export function mungOptionsForFetch(_options) {
+  const options = assign({
+    credentials: 'same-origin',
+  }, _options);
+
+  options.method = options.type;
+
+  // GET and HEAD requests can't have a `body`
+  if (options.data && Object.keys(options.data).length) {
+    if (options.method === 'GET' || options.method === 'HEAD') {
+      options.url += `?${serialiazeQueryParams(_options.data)}`;
```

Comment:
I found one error. This does not serialize nested objects properly.

Example:

`serialiazeQueryParams(_options.data)` with this implementation generates `page=%5Bobject%20Object%5D`.

The expected queryParams that is generated by `$.ajax` is `page%5Bnumber%5D=1&page%5Bsize%5D=3`